packages/core/useShare/index.ts

```diff
@@ -10,16 +10,24 @@ interface NavigatorWithShare {
   canShare?: (data: ShareOptions) => boolean
 }
 
-export async function useShare(shareOpts: ShareOptions) {
+export function useShare() {
```

Comment:
Oh I am thinking this, maybe we can allow users to pass a ref or reactive data to the constructor and then calling share to use the current state, what do you think?

```ts
const options = ref<ShareOptions>({ url: '...' })

return { options, ...useShare(options) }
```

```html
<input v-model='options.url'/>
<button @click='share()'>Share</button>
````

Reply:
Good idea!

Reply:
🤔    but thinking again hehe,  the users may also want to use it more programmatically, perhaps using it a little more complex. For example:

```js
const shareContentX =() = share({ ... })
const shareContentZ =() = share({ ... })
const shareContentZ = () = share({ ... })

// or

const shareContent = payload => share(payload)

if (...) shareContent(...)
else if (...) shareContent(...)
else shareContent(...)

// ...
```

Changing to the constructor he will have to manage the shared data in a single place (ref).

I haven't used this feature in my projects yet, so my thinking may be wrong  😬

Reply:
I am thinking like this, the returned share can accept an argument that overrides the "defaults"

```ts
const options = ref<ShareOptions>({ foo: 'bar' })

const { share } = useShare(options)

share() // { foo: 'bar' }

// change the source to change the payload. In the real-world, this can be a computed what dynamics on user content
options.value = { hello: 'world' }

share() // { hello: 'world' }

// temporary override payload
share({ url: 'github.com', text: 'hello world' }) // { hello: 'world', url: 'github.com', text: 'hello world' }

share() // { hello: 'world' }
```

This might cover most of the use cases I guess?

Reply:
Yes, it is a good solution and it might cover most use cases. Thanks!

Reply:
@antfu , I tried using permissions but was unsuccessful. It looks like we don't have support yet.